### PR TITLE
9796: QC Display: Add Date/Place of Trial to Calendared Work Items

### PR DESCRIPTION
**Pull request by shumway-tm**
- Display the date and place of trial session on the QC display when the case the work item is attached to is Calendared.

#### My Inbox

image

---

#### My Processed (Outbox)

717BD00E-BA50-4D0B-9BA7-7CD2C40CFB93

---

#### Section Inbox

A1BBC275-4931-4519-A092-689EA6DA7AF5

---

#### Section Processed (Outbox)

42BE036C-3507-4B67-AED8-0FB3F63A3631

---

#### Section In Progress

ED223278-95BE-462F-9181-600823F3EDA1

### CODE REVIEW

File: shared/src/persistence/dynamo/workitems/updateAttributeOnDynamoRecord.ts
Comment (rosiehsmith) [inline shared/src/persistence/dynamo/workitems/updateAttributeOnDynamoRecord.ts:1]: ❤️

File: shared/src/business/entities/OutboxItem.ts [hunk 1]
```diff
@@ -22,7 +22,8 @@ export class OutboxItemClass {
   public leadDocketNumber: string;
   public section: string;
   public assigneeId: string;
-  public trialDate: string;
+  public trialDate?: string;
```
Comment (rosiehsmith) [inline shared/src/business/entities/OutboxItem.ts:25]: trialDate is optional on OutboxItem now?

File: web-api/elasticsearch/efcms-work-item-mappings.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ module.exports = {
       type: 'join',
     },
     'completedAt.S': {
-      type: 'text',
+      type: 'date',
```
Comment (rosiehsmith) [inline web-api/elasticsearch/efcms-work-item-mappings.js:16]: Nice.

**Review by rosiehsmith — APPROVED** [review]

File: shared/src/business/entities/OutboxItem.ts (hunk 1 above)
Comment (shumway-tm) [reply shared/src/business/entities/OutboxItem.ts:25]: This was more to bring it in line with this field being optional on a WorkItem based on the assumption that an OutboxItem is simply a completed WorkItem. If that is incorrect I am happy to change it back!

Comment (shumway-tm) [reply shared/src/business/entities/OutboxItem.ts:25]: Also on further thought, I believe it is optional and allow(null) in the Joi Validation Rules if that makes a difference.

Comment (rosiehsmith) [reply shared/src/business/entities/OutboxItem.ts:25]: Makes sense! Thanks for clarifying that.

**Pull request merged**